File: DesignPatterns/6-mementoPattern/golang/memento.go
/*
 * File: memento.go
 * Created Date: 2023-03-19 02:32:35
 * Author: ysj
 * Description:  备忘录模式-备忘录(状态保存者)
 */
package main

type Memento struct {
	Skills  string // 技能
	HP      int    // 血量
	MP      int    // 蓝量
	Attack  int    // 攻击力
	Defense int    // 防御力
}

File: DesignPatterns/6-mementoPattern/python/memento.py
#!/usr/bin/env python3
# -*- coding:utf-8 -*-
###
# File: memento.py
# Created Date: 2023-03-19 03:30:40
# Author: ysj
# Description:  备忘录模式-备忘录(状态保存者)
###

class Memento(object):
    def __init__(
        self,
        skills="大威天龙",
        hp=100,
        mp=100,
        attack=100,
        defense=100,
    ):
        self.skills = skills
        self.hp = hp
        self.mp = mp
        self.attack = attack
        self.defense = defense
